CARDS-1441: Upgrade material ui and mui/lab to version 5+

veronikaslc opened the pull request:
**Testing instructions**
* **All functionality** in **all projects** on **all screen sizes** needs to be tested
* Due to mui v5 changes, gray color for buttons is no longer supported, and the default color is primary (dark blue). We replaced the gray look with outline variant. Check that button color/variant is appropriate across the app (all Close/Cancel buttons should be outlined).
  * Note that some buttons have intentionally changed color (e.g. "Healthcare provider" on the DATA-PRO landing page, "Decline" in the DATA-PRO Terms of use dialog)

**Known issues**
* The properties listing appears too condensed in the questionnaire wizard - not a blocker, can be adjusted as part of 1767.

Commit:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
adapter-v4 codemod Imports and inserts adaptV4Theme into createTheme (or createMuiTheme)

Commit:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
button-color-prop codemod - Removes the outdated color prop values.

Commit:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
bcreate-theme codemod - Renames the function createMuiTheme to createTheme.

Commit:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
dialog-props codemod - Remove disableBackdropClick prop from Dialog.

Commit:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
dialog-title-props codemod - Remove disableTypography prop from <DialogTitle>.

Commit:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
icon-button-size codemod - Adds size=large if size is not defined to get the same appearance as v4.

Commit:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
link-underline-hover codemod - Apply underline=hover to <Link /> that does not define underline prop (to get the same behavior as in v4).

Commit:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
material-ui-styles codemod - Moves JSS imports to @material-ui/styles

Commit:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
moved-lab-modules codemod - Updates all imports for @material-ui/lab components that have moved to @material-ui/core.

Commit:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
styled-engine-provider codemod - Applies StyledEngineProvider to the files that contains ThemeProvider.

Commit:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
table-props codemod - Renames props in Table* components.

Commit:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
theme-breakpoints codemod - Updates breakpoint values to match new logic.

Commit:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
theme-provider codemod - Renames MuiThemeProvider to ThemeProvider.

Commit:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
theme-spacing codemod - Removes the 'px' suffix from some template strings.

Commit:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
use-transitionprops codemod - Updates Dialog, Menu, Popover, and Snackbar to use the TransitionProps prop to replace the onEnter* and onExit* props.

Commit:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
variant-prop codemod - Adds the TextField, Select, and FormControl's variant=standard prop when variant is undefined.

Commit:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
mui-replace codemod - Replace every occurrence of material-ui related package with the new package names.

Commit:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
Dependencies upgreade

Commit:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
Import Timeline components from the lab

Commit:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
To avoid a type error when passing ref use a specific element type.

Commit:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
Removed unused artifact resource

Commit:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
Rolling back babel version upgrade

Commit:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
Fixed theme provider to be at the top component level

Commit:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
Fixed Popper React internal dependency upgraded from v1 to v2

Commit:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
 Rolling back unnecessary dependecies versions upgrades

Commit:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
 Rolling back accidental meaningless java file changes

Commit:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
 Rolling back white space changes

Commit:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
Rolling back unnecessary dependecies versions upgrades

Commit:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
 Rolling back white space changes

Commit:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
Removed non-existent formatDate property from TextInput

Commit:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
Some more component migration changes

veronikaslc committed:
Merge branch 'dev' into CARDS-1441-simplified

marta- committed:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
Updated the styling of the ResourceHeader component to account for the new way Grid items are spaced in Mui5

marta- commented:
> * The ResourceHeader (used on form pages and questionnaire pages) is now shifted to the top-left and looks misaligned with the rest of the page

Fixed in 9cc23c1.

marta- committed:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
Updated the default size of the Delete and Print buttons

marta- commented:
> * Livetables: the action buttons Edit and Delete now have different sizes and are misaligned
> * Subject pages: the menu looks too small/crowded due to the buttons being of size medium instead of large

Fixed in dd12d0e.

marta- commented on modules/commons/src/main/frontend/src/components/UserInputAssistant.jsx:
```diff
@@ -127,12 +128,12 @@ function UserInputAssistant (props) {
           <CardActions>
           { actionLabel && onAction &&
-            <Button color="default" onClick={onAction}>{actionLabel}</Button>
+            <Button onClick={onAction}>{actionLabel}</Button>
```
```suggestion
            <Button variant="outlined" onClick={onAction}>{actionLabel}</Button>
```

marta- commented on modules/commons/src/main/frontend/src/components/UserInputAssistant.jsx:
```diff
@@ -127,12 +128,12 @@ function UserInputAssistant (props) {
           }
           { onIgnore ?
-            <Button color="default" onClick={() => {setEnabled(false); onIgnore();}}>Ignore for now</Button>
+            <Button onClick={() => {setEnabled(false); onIgnore();}}>Ignore for now</Button>
```
```suggestion
            <Button variant="outlined" onClick={() => {setEnabled(false); onIgnore();}}>Ignore for now</Button>
```

marta- commented on modules/commons/src/main/frontend/src/components/UserInputAssistant.jsx:
```diff
@@ -127,12 +128,12 @@ function UserInputAssistant (props) {
+            <Button onClick={() => {setEnabled(false); onIgnore();}}>Ignore for now</Button>
             :
-            <Button color="default" onClick={() => {setEnabled(false)}}>Got it!</Button>
+            <Button onClick={() => {setEnabled(false)}}>Got it!</Button>
```
```suggestion
            <Button variant="outlined" onClick={() => {setEnabled(false)}}>Got it!</Button>
```

marta- commented on modules/data-entry/src/main/frontend/src/dataHomepage/Filters.jsx:
```diff
@@ -572,7 +575,6 @@ function Filters(props) {
           </Button>
           <Button
             variant="contained"
```
```suggestion
            variant="outlined"
```

marta- commented on modules/data-entry/src/main/frontend/src/dataHomepage/NewFormDialog.jsx:
```diff
@@ -399,7 +399,6 @@ function NewFormDialog(props) {
           }
           <Button
             variant="contained"
```
```suggestion
            variant="outlined"
```

marta- commented on modules/data-entry/src/main/frontend/src/dataHomepage/UserDashboard.jsx:
```diff
@@ -130,7 +130,6 @@ function UserDashboard(props) {
         <DialogActions>
           <Button
             variant="contained"
```
```suggestion
            variant="outlined"
```

marta- commented on modules/data-entry/src/main/frontend/src/questionnaire/Section.jsx:
```diff
@@ -300,7 +301,6 @@ function Section(props) {
           <Button
             size="small"
             variant="contained"
```
```suggestion
            variant="outlined"
```

marta- commented on modules/data-entry/src/main/frontend/src/questionnaire/SubjectSelector.jsx:
```diff
@@ -142,7 +143,6 @@ function UnstyledNewSubjectDialog (props) {
           <Button
             onClick={() => {setNewSubjectType(""); onClose()}}
             variant="contained"
```
```suggestion
            variant="outlined"
```

marta- commented on modules/data-entry/src/main/frontend/src/questionnaire/SubjectSelector.jsx:
```diff
@@ -271,15 +271,13 @@ function UnstyledSelectParentDialog (props) {
         <Button
           onClick={onClose}
           variant="contained"
```
```suggestion
          variant="outlined"
```

marta- commented on modules/data-entry/src/main/frontend/src/questionnaire/SubjectSelector.jsx:
```diff
@@ -271,15 +271,13 @@ function UnstyledSelectParentDialog (props) {
         </Button>
         <Button
           onClick={onBack}
           variant="contained"
```
```suggestion
          variant="outlined"
```

marta- commented on modules/data-entry/src/main/frontend/src/questionnaire/SubjectSelector.jsx:
```diff
@@ -693,7 +691,6 @@ function UnstyledSelectorDialog (props) {
         <Button
           onClick={onClose}
           variant="contained"
```
```suggestion
          variant="outlined"
```

marta- commented on modules/data-entry/src/main/frontend/src/questionnaireEditor/NewQuestionnaireDialog.jsx:
```diff
@@ -126,7 +128,6 @@ function NewQuestionnaireDialog(props) {
           </Button>
           <Button
             variant="contained"
```
```suggestion
            variant="outlined"
```

marta- commented on modules/data-entry/src/main/frontend/src/vocabQuery/VocabularyTree.jsx:
```diff
@@ -277,8 +278,7 @@ function VocabularyTree(props) {
-          <Button color="default"
-                  onClick={onCancel}
+          <Button onClick={onCancel}
                   variant="contained"
```
```suggestion
                  variant="outlined"
```

marta- commented on modules/login/src/main/frontend/src/login/loginMainComponent.js:
```diff
@@ -49,42 +48,39 @@ class MainLoginContainer extends React.Component {
+            </Grid>
+            { this.state.isLongForm && (!this.state.signInShown || this.state.signUpEnabled) &&
+              <Grid item>
+                <Button
```
```suggestion
                <Button
                  variant="outlined"
```

marta- commented on modules/statistics/src/main/frontend/src/Statistics/AdminStatistics.jsx:
```diff
@@ -286,7 +286,6 @@ function StatisticDialog(props) {
         <Button
             onClick={onClose}
             variant="contained"
```
```suggestion
            variant="outlined"
```

marta- commented on proms-resources/frontend/src/main/frontend/src/proms/Clinics.jsx:
```diff
@@ -209,7 +209,6 @@ function OnboardNewClinicDialog(props) {
           <Button
             onClick={onClose}
             variant="contained"
```
```suggestion
            variant="outlined"
```

marta- commented on proms-resources/frontend/src/main/frontend/src/proms/PromsLandingPage.jsx:
```diff
@@ -77,60 +75,58 @@ function PromsLandingPage(props) {
+                    <Button
+                      fullWidth
+                      variant="contained"
+                      color="primary"
```
```suggestion
```

marta- commented on proms-resources/frontend/src/main/frontend/src/proms/ToUDialog.jsx:
```diff
@@ -179,7 +181,7 @@ function ToUDialog(props) {
             Accept
           </Button>
-          <Button color="default" onClick={() => setShowConfirmationTou(true)} variant="contained" >
+          <Button onClick={() => setShowConfirmationTou(true)} variant="contained" >
```
```suggestion
          <Button variant="outlined" onClick={() => setShowConfirmationTou(true)} variant="contained" >
```

marta- commented on proms-resources/frontend/src/main/frontend/src/proms/ToUDialog.jsx:
```diff
@@ -202,7 +204,7 @@ function ToUDialog(props) {
             Accept
           </Button>
-          <Button color="default" onClick={() => {setShowConfirmationTou(false); onDecline && onDecline()}} variant="contained" >
+          <Button onClick={() => {setShowConfirmationTou(false); onDecline && onDecline()}} variant="contained" >
```
```suggestion
          <Button variant="outlined" onClick={() => {setShowConfirmationTou(false); onDecline && onDecline()}} variant="contained" >
```

marta- reviewed (commented):
All the`<Button>`s that had `color="default"` in Mui v.4 should have `variant="outlined"` in Mui v.5.

marta- committed:
CARDS-1441: Upgrade material ui and mui/lab to version 5+

All the <Button>s that had color="default" in Mui v.4 should have variant="outlined" in Mui v.5.
Exception: the "Healthcare Provider" button on the DATA-PRO landing page should have color="primary" (now default in v5) just like the "Patient" button above it.

marta- commented:
> * A few buttons that used to have the default (gray) color now have the primary (midnight blue) color.

Addressed in 32e66e8 by setting `variant="outlined"` to all the buttons that had `color="default"` removed in previous commits on this branch. There may be buttons that did not have `color="default"` explicitly specified. Those need to be detected and updated when testing the entire UI.

marta- committed:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
Update the imports in QuestionMatrix.jsx to match mui 5+

marta- commented:
> * **[MAJOR]** On forms with QuestionMatrix entries (e.g. SF-12 in `-P proms` or Question Matrix Test in `--test`) there seems to be some strange kind of styling failure

Fixed in 8dbd8f5 by updating the imports in omitted file `QuestionMatrix.jsx` to mui 5.

marta- committed:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
Updated two omitted buttons which still had `color="default"`, causing js console errors and white screens.

marta- commented:
> * **[MAJOR]** Attempting to edit a Concept Recognition Test form fails with white screen and console error
> * **[BLOCKER]** Attempting to add a question in a new questionnaire in the designer fails with white screen and console error
> * **[BLOCKER]** Attempting to move on to the second page of a "Pagination Test" form fails with white screen and console error

Fixed in c7d76b9 by updating two `<Button>`s which still had `color="default"`, causing js console errors and white screens.

Commit:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
Renamed deprecated buttonRef prop to ref (changemade for ButtonBase and propagates to InfoButton component)

veronikaslc commented:
`[BLOCKER] Clicking on the info buttons when browsing vocabulary terms no longer displays the info box`

Fixed in https://github.com/data-team-uhn/cards/pull/1045/commits/afa631975a9b3858eda73f5f03cdaab2e3ff770d

Commit:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
Revert unrelated whitespace changes

marta- committed:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
Set `variant="outlined"` to all buttons that used to have default color in mui v4.
Also adjust some other button colors (from secondary to primary) for consistency across the platform ui.

marta- committed:
CARDS-1441: Upgrade material ui and mui/lab to version 5
Reverted the variant of some `TextField`s that were explicitly set to "outlined" before the upgrade, but got converted to "standard" by mistake.

marta- committed:
CARDS-1441: Upgrade material ui and mui/lab to version 5
Updated the omitted `downtime-warning-banner` to mui v5.

marta- commented:
> * A few buttons that used to have the default (gray) color now have the primary (midnight blue) color.

Set `variant="outlined"` to all buttons that didn't have a color set (defaulting to gray in v4) in 79ea6da.

> * Some text fields have turned from underlined to outlined (vocabulary table filters) or from outlined to underlined (install vocabulary from local file). We should decide which variant to settle on.

Reverted the subject identifier input (in the Subject selector dialog) and vocab keyword search input (in the vocab admin screen) to `outlined` in 63d1735. The inputs in the form for installing a vocab from a local file look fine with `variant="standard"`. The inputs for vocabulary table filters should be reverted, but appear trickier to change at this time so we'll leave them as they are for the time being.

Also upgraded the `downtime-warning-banner` module (brought in by dev merge-in but ommitted from upgrade) to mui v5 in 2021032.

marta- committed:
CARDS-1441: Upgrade material ui and mui/lab to version 5
DATA-PRO Patient UI: Adjusted the colors and styles of the Terms of Use dialog buttons

marta- commented:
> The inputs for vocabulary table filters should be reverted, but appear trickier to change at this time so we'll leave them as they are for the time being.

Implemented simpler-looking filter inputs in a925672.

marta- committed:
CARDS-1441: Upgrade material ui and mui/lab to version 5
Use a simpler-looking input for vocab table filters on the Vocabularies admin page

marta- commented:
There is one known outstanding issue, which is not a blocker:

> * Sections with displayMode: header are sticky to a position slightly lower than they should be

@sashaandjic This branch is now ready for testing.

marta- committed:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
Fix positioning of sticky header sections.
The positioning was inaccurate even before the upgrade, but the change in how grid items are spaced within the container in 5+ made the issue more prominent post upgrade.

marta- commented:
> * Sections with displayMode: header are sticky to a position slightly lower than they should be

Fixed in 960b839. The positioning was inaccurate even before the upgrade, but the change in how `Grid item`s are spaced within the `Grid container` in 5+ made the issue more prominent post upgrade.

marta- committed:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
* Two more TextFields were left with outlined variant instead of standard in AnswerOptions
* Also fixed the positioning of the `enable` switches for na/noneOfTheAbove

Commit:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
Change deprecated Hidden component to usage of sx components prop for breakpoints control

Commit:
Merge branch 'CARDS-1441-simplified' of https://github.com/data-team-uhn/cards into CARDS-1441-simplified

marta- committed:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
Bug fix: search box and the user menu are always displayed on narrow screens

marta- commented:
> BLOCKER: The narrow-screen layout is slightly broken - the user avatar and menu are visible even though they should not be

Fixed in [a8e7282](https://github.com/data-team-uhn/cards/pull/1045/commits/a8e72824dbd6317d0b1efc8d71e1d86b30723487) and [9734509](https://github.com/data-team-uhn/cards/pull/1045/commits/973450931df9d9907905a9373c936ea8c0ee0c56).

sashaandjic commented:
a small inconsistency:
<img width="308" alt="image" src="https://user-images.githubusercontent.com/23243244/172468986-e3ed2350-1530-41cb-b4d7-74de8d7bde91.png">

marta- committed:
CARDS-1441: Upgrade material ui and mui/lab to version 5+

Fix the color of "Decline" button in the proms Terms of Use confirmation dialog

marta- commented:
> a small inconsistency

Fixed in e8560cf .

sashaandjic commented:
The buttons have different alignments
![image](https://user-images.githubusercontent.com/23243244/172471328-d9dc7275-f22f-44bf-a54d-fe82a81365f8.png)

marta- committed:
CARDS-1441: Upgrade material ui and mui/lab to version 5+

Updated the BACK button style in FormPagination, for consistency with the rest of the UI

sashaandjic commented:
<img width="497" alt="image" src="https://user-images.githubusercontent.com/23243244/172475907-1a575d26-e153-4203-8402-b774d7f23331.png">

marta- commented:
The last one is intentional. Just because our secondary color happens to currently be a a shade of red doesn't mean it's right to make DELETE a secondary action. However, now in mui5+ it's easy to use other theme colors for buttons, e.g. `"error"` if we wish to make it red so that we draw attention to it. @sashaandjic do you want do do that? I'm fine either way.

marta- committed:
CARDS-1441: Upgrade material ui and mui/lab to version 5+

One more button (in PrintPreview) that had the default color in mui4 and should be switched to `variant="outlined"` in mui5.

marta- commented:
PrintPreview CLOSE button fixed in 5c1aa3a.

marta- committed:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
Use `error` color to emphasize DELETE buttons

marta- commented:
> The last one is intentional. Just because our secondary color happens to currently be a a shade of red doesn't mean it's right to make DELETE a secondary action. However, now in mui5+ it's easy to use other theme colors for buttons, e.g. `"error"` if we wish to make it red so that we draw attention to it. @sashaandjic do you want do do that? I'm fine either way.

Introduced `error` color for DELETE buttons in f46d612.

marta- committed:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
More button appearance fixes:
* Adjusted the size of the DeleteButton in the Form's "More actions" menu to match that of the other menu entries
* Updated the styling of the visit selection buttons in Proms UI to match the new mui5 css classes

marta- commented:
> The buttons have different alignments ( dev left; 1441 center)

Fixed in 1238252.

marta- committed:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
Adjusted the look of sliders:
* made the value bubble red
* removed styling that no longer applies to the new html structure of sliders

marta- commented:
Sliders adjusted in 1a35d16.

sashaandjic reviewed (approved):
tested

sdumitriu commented on modules/commons/src/main/frontend/src/components/ResponsiveDialog.jsx:
```diff
@@ -98,7 +99,6 @@ const ResponsiveDialog = forwardRef((props, ref) => {
       maxWidth={width}
       fullWidth
       fullScreen={fullScreen}
-      disableBackdropClick
```
So, no more preventing dialogs from being closed when clicking outside?

marta- replied:
https://mui.com/material-ui/migration/v5-component-changes/#modal

The `disableBackdropClick` was removed in mui5. We can still implement the functionality by checking the `reason` in the `onClose` handler.

marta- replied:
Done in 61464d6.

sdumitriu commented on modules/data-entry/src/main/frontend/src/dataHomepage/PermissionsButton.jsx:
```diff
@@ -19,8 +19,9 @@
 import React from "react";
```
Huh, didn't know we have a permissions button. Is it used anywhere?

marta- replied:
Not used (yet). It was developed a long time ago as part of a mockup.

sdumitriu commented on modules/data-entry/src/main/frontend/src/questionnaire/NumberQuestion.jsx:
```diff
@@ -68,10 +69,8 @@ const useSliderStyles = makeStyles(theme => ({
-          transform: "rotate(135deg)",
-        },
+        background: theme.palette.secondary.main,
+        transform: "rotate(-90deg) translate3d(7px, -18px, 0) !important",
```
I prefer the way the slider looked before; is the change in appearance intentional?

marta- replied:
They changed the default look in mui (from bubble to callout). The callout doesn't scale well when rotated. I'd leave it as is, it's robust and looks decent imo.

sdumitriu commented on modules/data-entry/src/main/frontend/src/dataHomepage/PrintButton.jsx:
```diff
@@ -20,8 +20,9 @@ import React, { useState, useRef, useEffect } from "react";
 import { withRouter } from "react-router-dom";
 import PropTypes from "prop-types";
 
-import { Button, IconButton, Tooltip, withStyles } from "@material-ui/core";
```
Not sure if this is the right component, but the "Print preview" entry in the form action menu looks different from the other two entries: it's bigger and offset:

![image](https://user-images.githubusercontent.com/88663/172512697-ac694b0c-08e8-41ec-8df5-4baaf7cdae2c.png)

Also, the menu entries weren't colored before, is this color change also intentional?

marta- replied:
The color change is intentional. I will fix the button size.

veronikaslc replied:
Fixed in https://github.com/data-team-uhn/cards/pull/1045/commits/85b113470214232fbbd548dac112b7938093b49a

marta- replied:
Completely fixed in e4d1e78.

sdumitriu commented on modules/data-entry/src/main/frontend/src/questionnaire/ResourceHeader.jsx:
```diff
@@ -128,20 +124,16 @@ function ResourceHeader (props) {
           { fullBreadcrumbTrigger && <div className={classes.breadcrumbAction}>{action}</div> }
         </Collapse>
       </Grid>
-      <Collapse in={fullBreadcrumbTrigger}>
```
How is this still working without the collapse? Or is this not the thing that makes the form header collapse when scrolling?

marta- replied:
This part of the header just scrolls up with the rest of the form.

sdumitriu commented on modules/pedigree/src/main/frontend/src/pedigree/style/phenotipsWidgets.css:
```diff
@@ -598,7 +598,6 @@ input.xwiki-free-multiselect-value {
  margin:.4em 0
 }
 input.suggestWorkgroups {
- background-image:url('/resources/icons/datamodel/groups.png');
```
Looks like we may have some broken resources in the pedigree editor. Are there any other? Should we try to fix the missing icon?

marta- replied:
This is the only problematic icon in the pedigree editor. The functionality using this icon is not even exposed in the editor within cards. I would leave further cleanup of the pedigree for later.

sashaandjic commented:
Very small thing: there seem to be a blank before `Print preview`.
![image](https://user-images.githubusercontent.com/23243244/172535726-f2bb6c7b-e3cb-40b1-a7cb-759dc48d151c.png)

sashaandjic commented:
Another mini issue with the alignment:
![image](https://user-images.githubusercontent.com/23243244/172535953-3bc46c8a-9d51-4e75-ad8c-b460623655b6.png)

Commit:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
Fixed button size for printing and export buttons

Commit:
Merge branch 'CARDS-1441-simplified' of https://github.com/data-team-uhn/cards into CARDS-1441-simplified

veronikaslc commented:
Fixed print preview in https://github.com/data-team-uhn/cards/pull/1045/commits/85b113470214232fbbd548dac112b7938093b49a

sashaandjic commented:
fixed:
![image](https://user-images.githubusercontent.com/23243244/172563547-dfc8b681-bc85-495f-988c-7fb50b033672.png)
![image](https://user-images.githubusercontent.com/23243244/172564401-ab49e3a7-10e9-4ed0-805a-aa6dbc4a1318.png)

one more small misalignment:
![image](https://user-images.githubusercontent.com/23243244/172564544-f59abbcd-8812-4813-bfdb-121f86ce1b28.png)

marta- committed:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
Changed the look of the Review Terms button in Proms for consistency with the rest of the ui

marta- committed:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
Replace the disableBackdropClick prop, removed in mui5, with checking the "reason" in the "onClose" handler.

marta- committed:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
Made button sizes consistent across the Form's dropdown menu in view mode

marta- commented:
> fixed: ![image](https://user-images.githubusercontent.com/23243244/172563547-dfc8b681-bc85-495f-988c-7fb50b033672.png) ![image](https://user-images.githubusercontent.com/23243244/172564401-ab49e3a7-10e9-4ed0-805a-aa6dbc4a1318.png)

Nope, now the Print and Export buttons are both large, while Delete is medium.

marta- commented:
> > fixed

> Nope, now the Print and Export buttons are both large, while Delete is medium.

Sizes are now consistent since 7d089cf.

marta- committed:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
Adjust the styling of Proms Patient UI screens to account for how
Grid item spacing within the container has changed in mui5.

marta- commented:
> one more small misalignment: 

Fixed in 70c43dd.

marta- committed:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
Use the "success" color (newly supported for buttons in mui5) for the New subject button
in the New Form Dialog / Subject selector. The currenly used color ("secondary")
accidentally looks a bit alarming.

marta- commented:
Additional changes not requested in the comments:
* Proms UI. Terms of Use dialog: The "Review Terms" button is now `primary outlined` instead of `secondary contained`, to be consistent with other "neutral" actions across the platforms - commit 5aae60d
* The "New subject" button now has the `success` color instead of `secondary`, to make it look less alarming and more like an `Add` action - commit 7cffef1

sashaandjic commented:
this one is now really fixed :)
![image](https://user-images.githubusercontent.com/23243244/172649198-ea78887b-3fcf-4c4f-9dae-a434d216c792.png)

and this one:
![image](https://user-images.githubusercontent.com/23243244/172649310-9b56af05-61b0-486e-b80c-f9dcf78f14b6.png)

marta- committed:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
Fix broken layout of custom pagination for MatierialTables on the Subject page

marta- committed:
CARDS-1441: Upgrade material ui and mui/lab to version 5+
Temporary fix for the duplicate displayedRows occurring in material-table 2.0.3

marta- committed:
Merge branch 'dev' into CARDS-1441-simplified

sashaandjic reviewed (approved):
tested - all good
![image](https://user-images.githubusercontent.com/23243244/172699389-ff840623-0258-4664-bfe5-4fd515014777.png)
![image](https://user-images.githubusercontent.com/23243244/172699412-22a44c18-bd5a-4a9f-876c-0910a12db47a.png)
![image](https://user-images.githubusercontent.com/23243244/172699522-bc35bfbb-2b55-41c3-aaef-df3a7a22d6e9.png)
